upup/models/cloudup/resources/addons/networking.projectcalico.org.canal/k8s-1.15.yaml.template

```diff
@@ -422,31 +456,24 @@ subjects:
   name: canal
   namespace: kube-system
 ---
-# Bind the Calico ClusterRole to the canal ServiceAccount.
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: canal-calico
+  labels:
+    role.kubernetes.io/networking: "1"
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: calico
+  name: calico-node
 subjects:
 - kind: ServiceAccount
   name: canal
   namespace: kube-system
 
+{{ if .Networking.Canal.TyphaReplicas -}}
 ---
-
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: canal
-  namespace: kube-system
```

Comment:
So this one has puzzled me. I would have expected that the ServiceAccount must exist before the deployment that references it is created, but I guess as all resources are applied in quick succession the SA will be created before the scheduler kicks in to create the DS pods.

Reply:
I did not see any issue with this. I agree that this is the recommended way to do it, but it would be different from the official manifest and make it harder to compare with official manifests.
I would first try to push the change in official manifests.